pkg/alertmanager/alertmanager.go

```diff
@@ -188,19 +188,25 @@ func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
 	am.state = newReplicatedStates(cfg.UserID, cfg.ReplicationFactor, cfg.Replicator, cfg.Store, am.logger, am.registry)
 	am.persister = newStatePersister(cfg.PersisterConfig, cfg.UserID, am.state, cfg.Store, am.logger, am.registry)
 
-	am.wg.Add(1)
 	var err error
-	am.nflog, err = nflog.New(
-		nflog.WithRetention(cfg.Retention),
-		nflog.WithSnapshot(filepath.Join(cfg.TenantDataDir, notificationLogSnapshot)),
-		nflog.WithMaintenance(maintenancePeriod, am.stop, am.wg.Done, nil),
```

Comment:
Note to reviewers: this option doesn't exist anymore because now you're expected to start the maintenance gorouting (done below).